src/buildblock/ProjDataInMemory.cxx

```diff
@@ -350,6 +350,18 @@ ProjDataInMemory::ProjDataInMemory(const ProjDataInMemory& proj_data)
   std::copy(proj_data.begin_all(), proj_data.end_all(), this->begin_all());
 }
 
+shared_ptr<ProjDataInMemory>
+ProjDataInMemory::read_from_file(const std::string& filename, const std::ios::openmode openmode)
+{
+#if 1
+  const auto proj_data = ProjData::read_from_file(filename, openmode);
+  return std::make_shared<ProjDataInMemory>(*proj_data);
+#else
+  // Alternative implementation
+  return std::make_shared<ProjDataInMemory>(*ProjData::read_from_file(filename, openmode));
+#endif
```

Comment:
Not sure which implementation is best. I think the second

```cpp
return std::make_shared<ProjDataInMemory>(*ProjData::read_from_file(filename, openmode));
```
might prevent an additional copy. I am just not sure about the data being correctly loaded.

Reply:
they're identical really. it needs to make a copy of the data anyway as `PojData::read_from_file` will create a "read from disk" object.